client/src/components/common/applicant-info-nav.vue

```diff
@@ -56,53 +57,61 @@ export default class ApplicantInfoNav extends Vue {
     return newReqModule.nrState
   }
   get showBack () {
-    if (this.tab < 2) {
+    if (this.submissionTabNumber < 2) {
       return false
     }
-    if (this.tab === 2) {
+    if (this.submissionTabNumber === 2) {
       return (this.type === 'examination' || this.nrState === 'DRAFT')
     }
-    if (this.tab === 3) {
+    if (this.submissionTabNumber === 3) {
       return true
     }
     return false
   }
-  get tab () {
+  get submissionTabNumber (): number {
     return newReqModule.submissionTabNumber
   }
   get type () {
     return newReqModule.submissionType
   }
   back () {
-    newReqModule.mutateSubmissionTabNumber(this.tab - 1)
+    newReqModule.mutateSubmissionTabNumber(this.submissionTabNumber - 1)
   }
   async next () {
-    if (this.tab === 3) {
+    if (this.submissionTabNumber === 3) {
       this.isloadingSubmission = true
       await this.submit()
       return
     }
-    newReqModule.mutateSubmissionTabNumber(this.tab + 1)
+    newReqModule.mutateSubmissionTabNumber(this.submissionTabNumber + 1)
   }
 
+  /** Submits an edited NR or a new name submission. */
   async submit () {
-    let request
     // FUTURE: fix error handling in case of newReqModule (app or api) error (#5899)
     const { nrId } = this
     if (this.editMode) {
-      await newReqModule.patchNameRequests()
-      await newReqModule.checkinNameRequest()
+      // stop timer first so it doesn't expire during network requests
       timerModule.stopTimer({ id: this.$EXISTING_NR_TIMER_NAME })
-      await this.fetchNr(+nrId)
+      if (await newReqModule.patchNameRequests()) {
+        if (await newReqModule.checkinNameRequest()) {
+          newReqModule.mutateDisplayedComponent('Success')
+          await sleep(1000)
+          await this.fetchNr(+nrId)
+        }
```

Comment:
This is to display the "success" component for 1 second before fetching NR and loading another page.